pkgs/applications/office/morgen/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, stdenv, fetchurl, dpkg, autoPatchelfHook, makeWrapper, electron
+, nodePackages, alsa-lib, gtk3, libxshmfence, mesa, nss }:
+
+stdenv.mkDerivation rec {
+  pname = "morgen";
+  version = "2.3.5";
+
+  src = fetchurl {
+    url = "https://download.todesktop.com/210203cqcj00tw1/morgen-${version}.deb";
+    sha256 = "sha256-BSB2FRKtUjT+HhvetBg+dAn/Tilwbjph+LZgQM9simc=";
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    makeWrapper
+    nodePackages.asar
+  ];
+
+  buildInputs = [ alsa-lib gtk3 libxshmfence mesa nss ];
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  unpackPhase = ''
+    dpkg-deb -x ${src} ./
+  '';
```

Comment:
```suggestion
  unpackCmd = ''
    dpkg-deb -x ${src} ./
  '';

Reply:
Not clear of the obtained error:

```
·> nix-build -A morgen                                                                                     [0/161]
this derivation will be built:
  /nix/store/kv86q2snk9z8ylkanr2j21jc1aa8i10x-morgen-2.4.3.drv
building '/nix/store/kv86q2snk9z8ylkanr2j21jc1aa8i10x-morgen-2.4.3.drv'...
unpacking sources
unpacking source archive /nix/store/pm1sjk6q9zg4alqmnlwc23sh9lmz07d1-morgen-2.4.3.deb
unpacker produced multiple directories
error: builder for '/nix/store/kv86q2snk9z8ylkanr2j21jc1aa8i10x-morgen-2.4.3.drv' failed with exit code 1;
       last 3 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/pm1sjk6q9zg4alqmnlwc23sh9lmz07d1-morgen-2.4.3.deb
       > unpacker produced multiple directories
       For full logs, run 'nix log /nix/store/kv86q2snk9z8ylkanr2j21jc1aa8i10x-morgen-2.4.3.drv'
```

Reply:
Ok, it needs `sourceRoot`. Specifying it works.

Now, the question is, what is the main benefit of using `unpackCmd`? Just curious, because I generally use `unpackPhase` and never used `unpackCmd` before.

Reply:
> Now, the question is, what is the main benefit of using unpackCmd? Just curious, because I generally use unpackPhase and never used unpackCmd before.

I think it appends to `unpackPhase` instead over overriding it entirely, so the `preUnpack` and `postUnpack` hooks get ran without you having to type it all out. Realistically not running those won't really influence anything, but dependencies could potentially use them. Not a very important thing, but I think it's good practice to use.

My suggestion was a bit wrong though, it should've been something like:
```nix
  unpackCmd = ''
    dpkg-deb -x ${src} ./morgen-${version}
  '';
```

`sourceRoot` gets set automatically if there's a subdirectory, instead of extracting it to `.`

Reply:
The explanation makes a lot of sense, thank you. Maybe the tedious part is remembering the last bit of the unpack, but it is not an issue.

I have updated the code.

Reply:
Happy to help :)

I should add that the directory name you're unpacking to (`./morgen-${version}` here) doesn't really matter, but since it gets logged a few times I like it to include the package name and version.